src/main/java/org/prebid/server/bidder/adnuntius/AdnuntiusBidder.java

```diff
@@ -248,35 +248,33 @@ private List<HttpRequest<AdnuntiusRequest>> createHttpRequests(
     }
 
     private String makeEndpoint(BidRequest bidRequest, Boolean noCookies) {
-        try {
-            final String gdpr = extractGdpr(bidRequest.getRegs());
-            final String url = StringUtils.isNotBlank(gdpr) ? euEndpoint : endpointUrl;
+        final String gdpr = extractGdpr(bidRequest.getRegs());
+        final String url = StringUtils.isNotBlank(gdpr) ? euEndpoint : endpointUrl;
 
-            if (url == null) {
-                throw new PreBidException("an EU endpoint is required but invalid");
-            }
+        if (url == null) {
+            throw new PreBidException("an EU endpoint is required but invalid");
+        }
 
-            final URIBuilder uriBuilder = new URIBuilder(url)
-                    .addParameter("format", "prebidServer")
-                    .addParameter("tzo", getTimeZoneOffset());
+        final Map<String, String> queryParams = new HashMap<>();
 
-            if (StringUtils.isNotEmpty(gdpr)) {
-                uriBuilder.addParameter("gdpr", gdpr);
-            }
+        queryParams.put("format", "prebidServer");
+        queryParams.put("tzo", getTimeZoneOffset());
 
-            final String consent = extractConsent(bidRequest.getUser());
-            if (StringUtils.isNotEmpty(consent)) {
-                uriBuilder.addParameter("consentString", consent);
-            }
+        if (StringUtils.isNotEmpty(gdpr)) {
+            queryParams.put("gdpr", gdpr);
+        }
 
-            if (noCookies || extractNoCookies(bidRequest.getDevice())) {
-                uriBuilder.addParameter("noCookies", "true");
-            }
+        final String consent = extractConsent(bidRequest.getUser());
+        if (StringUtils.isNotEmpty(consent)) {
+            queryParams.put("consentString", consent);
+        }
 
-            return uriBuilder.build().toString();
-        } catch (URISyntaxException | IllegalArgumentException e) {
-            throw new PreBidException(e.getMessage());
+        if (noCookies || extractNoCookies(bidRequest.getDevice())) {
+            queryParams.put("noCookies", "true");
         }
+
+        return UriTemplate.of(url + (url.contains("?") ? "{&queryParams*}" : "{?queryParams*}"))
```

Comment:
Why are you creating template on each request, this is quite heavy operation? Template is thread safe and expansion is immutable - this can be reused. Hold template instead of `url` on class level and just expand when needed.